### PR TITLE
Apply not nice username checks

### PR DESCRIPTION
**Pull request by DanVanAtta**
This update prevents user from taking names contain a 'bad-word' entry are listed in the banned-username table.

commit 6addd4ece79f842d278eea17299deab362db59c5

    Validate usernames do not match banned usernames or bad words
    
    This update:
    - adds name validation to login, (it was already present
      for account creation). Name validation asserts that
      a given name does not contain a bad word.
    - adds banned username check to name validation. This check
      verifies that a given name does not match a banned username.

commit 628a59a6b584e098582e7587ce9d92f80cb8480c

    Split username is already taken check from login
    
    If we check if a username is already taken when doing
    a registered login, we will see the name is always taken!
    This update removes the username is already taken check
    from username validation and moves the check to anonymous
    login and user registration.

## Functional Changes

[] New map or map update
[] New Feature
[x] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[x] Problem fix: https://github.com/triplea-game/triplea/issues/6173 
[] Other:   

## Testing

Tested locally when registration and login are blocked based on data in banned usernames table or bad words (and was not previously)

## Additional Review Notes

Would like to comment that the extensive automated spitfire tests really helped here. I completely missed, and arguably it was not completely obvious, that the username is taken check was going to break registered logins.

### CODE REVIEW

File: http-server/src/main/java/org/triplea/modules/user/account/login/LoginModule.java [hunk 1]
```diff
@@ -108,7 +116,6 @@ private ApiKey recordAnonymousLoginAndGenerateApiKey(
     return recordLoginAndGenerateApiKey(loginRequest, systemId, playerChatId, ip, false);
   }
 
-  // TODO: Project#12 also update access log
   @SuppressWarnings("unused")
```
Comment (RoiEXLab) [inline http-server/src/main/java/org/triplea/modules/user/account/login/LoginModule.java:119]: Is this method still unused?

File: http-server/src/main/java/org/triplea/modules/user/account/login/authorizer/anonymous/AnonymousLoginFactory.java [hunk 2]
```diff
@@ -1,21 +1,6 @@
...
-
-  public static Function<UserName, Optional<String>> build(
-      final Jdbi jdbi, final Chatters chatters) {
-    return AnonymousLogin.builder()
-        .chatters(chatters)
-        .userJdbiDao(jdbi.onDemand(UserJdbiDao.class))
-        .build();
-  }
-}
+public class AnonymousLoginFactory {}
```
Comment (RoiEXLab) [inline http-server/src/main/java/org/triplea/modules/user/account/login/authorizer/anonymous/AnonymousLoginFactory.java:6]: I have the sneaking suspicion you forgot to remove this class 😉

File: http-server/src/main/java/org/triplea/modules/user/account/login/LoginModule.java (hunk 1 above)
Comment (DanVanAtta) [reply http-server/src/main/java/org/triplea/modules/user/account/login/LoginModule.java]: Looks like no, not sure how long the suppression was out of date for, looks like perhaps for some time.

File: http-server/src/main/java/org/triplea/modules/user/account/login/authorizer/anonymous/AnonymousLoginFactory.java (hunk 2 above)
Comment (DanVanAtta) [reply http-server/src/main/java/org/triplea/modules/user/account/login/authorizer/anonymous/AnonymousLoginFactory.java]: Twice even :grin: 
Was also missed earlier when we updated factory class pattern to be factory methods.

**Pull request merged**